pkg/storaged/content-views.jsx

```diff
@@ -70,30 +70,60 @@ function next_default_logical_volume_name(client, vgroup) {
     return name;
 }
 
+export function set_crypto_auto_option(block, flag, never) {
+    return edit_config(block, (config, commit) => {
+        const opts = config.options ? parse_options(utils.decode_filename(config.options.v)) : [];
+        if (opts.indexOf("x-cockpit-never-auto") >= 0)
```

Comment:
Yeah, thanks for putting your finger on this.  I _could_ explain it, but I'd rather try to figure out a way to get rid of it. :-)

(Roughly: mounting a unencrypted filesystem sets "auto" and unmounting sets "noauto", so that the "is-it-mounted" state persists through the next reboot. I want the same for encrypted filesystems. But, people have the use case where they explicitly do not want to mount a encrypted filesystem at boot, and only mount it manually as needed. With the current Cockpit behavior, mounting it manually would set "auto" and break the next boot. this use case came a bit late so I hacked up the x-never-auto option.)

Reply:
@mvollmer and I talked this over in detail in a meeting. I do buy the argument that the default behaviour of cockpit should be to keep persistent config and runtime state consistent. That's what we do on the Services and Firewall pages as well, and Storage aims to do that as well (see the warnings about inconsistencies that we already have). So `auto` and `noauto` in fstab are already written when doing Mount/Unmount.

There are two options:
 1. Always let the admin decide about temporary/permanently in the Mount dialog. Then some `x-cockpit-temp-mount` option would just be set for mount(1), and land in /run/mounts/utab (so that cockpit would not show the warning). After unmounting or rebooting, the system is in a consistent state again.
 2. If admins really want the "never, ever, period" config option, then that *is* a persistent and cockpit specific setting, which needs to be remembered in /etc. In that case I'm fine with `x-cockpit-never-auto`. **However:** This should be in /etc/fstab, not /etc/crypttab -- it is very plausible that we want the same for "normal" mounts, NFS, etc, and we really don't want to spread that option between two files.

Reply:
Related: issue #14467

Reply:
I have moved x-cockpit-never-auto to fstab in a WIP commit.  Still todo: Give it a real (i) info icon, there are no non-custom crypto options anymore, make it just a text field.

Reply:
All done now.